### PR TITLE
Tests For GetPathRoot and Enabling NormalizeDirectorySeparatorTests

### PR DESCRIPTION
**Pull request by Anipik**
Fixes https://github.com/dotnet/corefx/issues/27401

### CODE REVIEW

File: src/Common/tests/Tests/System/IO/PathInternal.Windows.Tests.cs [hunk 1]
```diff
@@ -141,5 +140,102 @@ public void AreRootsEqual(string first, string second, StringComparison comparis
...
+        };
+
+        public static TheoryData<string, int> GetRootLengthRooted => new TheoryData<string, int>
+        {
+            { @"\tmp", 1},
+            { @"\.", 1},
+            { @"\..", 1},
+            { @"\tmp\..\..", 1},
+            { @"\\", 2},
+            { @"\\\", 3},
```
Comment (JeremyKuhne) [inline src/Common/tests/Tests/System/IO/PathInternal.Windows.Tests.cs]: "\\\" should be probably be 2, but I'm not too concerned. I only care that initial double separators are considered special and are never coalesced into one separator. Having 3+ in the start is beyond what we want to handle or rationalize.

File: src/Common/tests/Tests/System/IO/PathInternal.Windows.Tests.cs [hunk 2]
```diff
@@ -141,5 +140,102 @@ public void AreRootsEqual(string first, string second, StringComparison comparis
...
+        {
+            { @"Server\Share\git\corefx", 13},
+            { @"Server\Share\git\.\", 13},
+            { @"Server\Share\git\..\", 13},
+            { @"Server\Share\git\..\..\", 13},
+            { @"Server\Share\..\", 13},
+            { @"Server\Share\", 13},
+            { @"Server\Share", 12},
+            { @"\\Server\Share", 14},
+            //{ @"\\Server\\Share", 15},
```
Comment (JeremyKuhne) [inline src/Common/tests/Tests/System/IO/PathInternal.Windows.Tests.cs]: Don't really care what we get for `\\\Server\Share`, `\\\\Server\\Share`, etc. It's good to test, but we're ok with a mess as Windows won't handle the paths well. You can use that as a metric- if GetFullPathName doesn't fix the separators correctly we can ignore it.

Bottom line is that up to the end of the "volume" I expect the # of separators provided to be correct. Outside of that, don't make the code any more complicated to deal with it.  (E.g. up to `C:` `C:\` `\\server\share`, `\\?\UNC\Server\Share` etc we should get the right number of separators.) Collapsing separators past that point is meant to be friendly to path concatenation.

File: src/Common/tests/Tests/System/IO/PathInternal.Windows.Tests.cs [hunk 3]
```diff
@@ -141,5 +140,102 @@ public void AreRootsEqual(string first, string second, StringComparison comparis
...
+        public static TheoryData<string, int> GetRootLength_DeviceData => new TheoryData<string, int>
+        {
+            { @"..\", 3},
+            { @"..\..\", 3},
+            { @"..\.\", 3},
+            { @"...\", 4},
+            { @".\", 2},
+            { @".\..\", 2},
+            { @"..\", 3},
+            { @"\.\", 3},
```
Comment (JeremyKuhne) [inline src/Common/tests/Tests/System/IO/PathInternal.Windows.Tests.cs]: I don't know if this is what we want. Ideally we want NormalizeDirectorySeparators to eat all "extra" separators once past the root and end up with a valid path that doesn't need normalized. `\\?\\C:\foo.txt` is an invalid path and cannot be opened. I believe `\\.\\C:\foo.txt` would get turned into `\??\C:\foo.txt` and work, but I'm not positive.

So I think device paths should give you just the length of the prefix if you hit a directory separator right after. I think that is a better choice, but if that makes things too complicated I wouldn't sweat it.

File: src/Common/tests/Tests/System/IO/PathInternal.Windows.Tests.cs (hunk 1 above)
Comment (Anipik) [reply src/Common/tests/Tests/System/IO/PathInternal.Windows.Tests.cs]: ```\\\``` is getting treated as a Unc path here. I am ignoring this input as we are expecting correct separators until volume

**Pull request merged**